### PR TITLE
Add the proper confidence in doc param checker's messages

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |

## Description

Add the confidence for the doc param checker and some minor refactors prior to #7212

### CODE REVIEW

File: pylint/extensions/docparams.py [hunk 1]
```diff
@@ -636,26 +645,23 @@ def _handle_no_raise_doc(self, excs: set[str], node: nodes.FunctionDef) -> None:
...
-            except KeyError:
-                pass
-
-        if not missing_excs:
-            return
-
-        self.add_message(
-            "missing-raises-doc", args=(", ".join(sorted(missing_excs)),), node=node
-        )
+            with contextlib.suppress(KeyError):
```
Comment (DanielNoord) [inline pylint/extensions/docparams.py:656]: I would advise against this as this is much worse for performance. `try...except` have minimal performance impact, especially in `3.11`. With context manager you need look up the manager in the module, set it up, call the `__enter__`, call the `__exit__`, etc. 
There was recent discussion about this in the `CPython` repository about the additional costs.

Comment (Pierre-Sassoulas) [reply pylint/extensions/docparams.py]: I reverted it, let's wait for the discussion on #7201to conclude before changing anything in pylint.

**Review by DanielNoord — APPROVED** [review]

File: pylint/extensions/docparams.py (hunk 1 above)
Comment (NeilGirdhar) [reply pylint/extensions/docparams.py]: @DanielNoord Could you link to the discussion please?

Comment (DanielNoord) [reply pylint/extensions/docparams.py]: https://github.com/python/cpython/issues/94933#issuecomment-1186581004

"Discussion" is perhaps too much 😅

**Pull request merged**